pkg/scheduler/actions/preempt/preempt.go

```diff
@@ -271,8 +271,16 @@ func preempt(
 
 		for !victimsQueue.Empty() {
 			// If reclaimed enough resources, break loop to avoid Sub panic.
-			// If preemptor's queue is overused, it means preemptor can not be allocated. So no need care about the node idle resource
-			if !ssn.Overused(currentQueue) && preemptor.InitResreq.LessEqual(node.FutureIdle(), api.Zero) {
+			// Preempt action is about preempt in same queue, which job is not allocatable in allocate action, due to:
+			// 1. cluster has free resource, but queue not allocatable
+			// 2. cluster has no free resource, but queue not allocatable
+			// 3. cluster has no free resource, but queue allocatable
+			// for case 1 and 2, high priority job/task can preempt low priority job/task in same queue;
+			// for case 3, it need to do reclaim resource from other queue, in reclaim action;
+			// so if current queue is not allocatable(the queue will be overused when consider current preemptor's requests)
+			// or current idle resource is not enougth for preemptor, it need to continue preempting
+			// otherwise, break out
+			if ssn.Allocatable(currentQueue, preemptor) && preemptor.InitResreq.LessEqual(node.FutureIdle(), api.Zero) {
```

Comment:
If the current capacity of the queue is 8C, 5C has been used, and the preemptor request is 4C, and the actual capacity of the queue exceeds 8C, allocatable should return false. The judgment logic here does not have a break, which is not in line with expectations.

I understand that the judgment logic here should be like this:
1. allocatable returns false and breaks directly
2. allocatable returns true, and then judges preemptor and futureIdle. 
    1) preemptor is less than or equal to futureIdle, execute break
    2) preemptor is less than or equal to futureIdle, break is not executed